### PR TITLE
Add execution timeouts to scheduling functions

### PR DESCRIPTION
**Pull request by skourta**
## TLDR

Introduce timeouts for schedule execution to enhance control over execution duration. This change allows specifying an execution timeout parameter in the `execute` function.

## Summary:
This pull request includes several changes to the `tiralib/tiramisu/compiling_service.py` and `tiralib/tiramisu/schedule.py` files, focusing on code formatting improvements and adding a new parameter for execution timeout.

### Code Formatting Improvements:
* [`tiralib/tiramisu/compiling_service.py`](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07L56-R58): Reformatted code blocks to improve readability by splitting long lines into multiple lines. [[1]](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07L56-R58) [[2]](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07L263-R267) [[3]](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07L421-R428) [[4]](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07L430-R449) [[5]](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07L569-R583)
* [`tiralib/tiramisu/schedule.py`](diffhunk://#diff-d8fa3bc539b578032c95949c268a80813b36566fc959caac876cf0f2c48d96efR127): Reformatted code blocks to improve readability by splitting long lines into multiple lines.

### New Parameter Addition:
* [`tiralib/tiramisu/compiling_service.py`](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07R399): Added `execution_timeout` parameter to the `get_cpu_exec_times` method and updated subprocess calls to include the timeout. [[1]](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07R399) [[2]](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07R497) [[3]](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07R526)
* [`tiralib/tiramisu/schedule.py`](diffhunk://#diff-d8fa3bc539b578032c95949c268a80813b36566fc959caac876cf0f2c48d96efR87): Added `execution_timeout` parameter to the `execute` method and updated the call to `get_cpu_exec_times` to pass this parameter. [[1]](diffhunk://#diff-d8fa3bc539b578032c95949c268a80813b36566fc959caac876cf0f2c48d96efR87) [[2]](diffhunk://#diff-d8fa3bc539b578032c95949c268a80813b36566fc959caac876cf0f2c48d96efR127)

### Exception Handling:
* [`tiralib/tiramisu/compiling_service.py`](diffhunk://#diff-e208dcee0560e47ba027b51b312d2843cc5b89bc0363781a6eb36c19e43ccc07L533-L534): Removed redundant exception handling code in the `get_cpu_exec_times` method.Introduce timeouts for schedule execution to enhance control over execution duration. This change allows specifying an execution timeout parameter in relevant functions.

### CODE REVIEW

**Review by Mascinissa — COMMENTED** [review]
Thanks for the PR. I had a different idea on how the execution budget should be handled please check [this issue](https://github.com/Tiramisu-Compiler/TiraLib/issues/36) and let me know if you have a different use case in mind.
